enclave/core/sgx/enter.S

```diff
@@ -119,6 +119,23 @@ oe_enter:
     mov %rsp, %rbp
 
 .call_function:
+    // Set the MXCSR according to the Linux x86_64 ABI
+    mov $ABI_MXCSR_INIT, %r10
+    push %r10
+    ldmxcsr (%rsp)
+    pop %r10
+
+    // Set the FPU Control Word according to the Linux x86_64 ABI
+    mov $ABI_FPUCW_INIT, %r10
+    push %r10
+    fldcw (%rsp)
+    pop %r10
+
+    // Initialize the RFLAGS prior to calling enclave functions
+    // This only clears the DF and state flag bits since
+    // the system flags and reserved bits are not writable here
+    push $0
```

Comment:
Do we need to save and restore RFLAGS register on the host side?

Reply:
Yes, it's cleaned out as part of `oe_cleanup_registers` on the enclave side on enclave exit, called as a part of an ocall or return from ecall. AEX handles the context cleanup in the exception case.

Reply:
To answer the "do we need to _save_" part of the question, no we don't. The RFLAGS are divided into:
* _Status flags_: regarded as volatile in both Linux and Windows ABIs
* _DF_: required to be cleared by both Linux and Windows ABIs
* _System Flags and IOPL Field_: privileged bits set by the OS, so we don't actually clear these when calling `popf` from user mode.